### PR TITLE
feat(sdk): reference existing secrets

### LINKED ISSUE #2252: `name` of `cloud.Secret` should refer to an existing name
### I tried this

```js
bring cloud;

new cloud.Secret(name: "my-secret-name");
```

### I expected this:

Since `name` refers to a physical name, I think it should mean that it references an *existing* secret instead of trying to create a new one.

If I want a new secret for every app, I expect this behavior:

```js
new cloud.Secret() as "my-secret-id";
```

And then `terraform apply` should output the `aws secretsmanager` command for putting the secret value:

```sh
aws secretsmanager put-secret-value --secret-id "my-secret-id-129823unb29438723" --secret-string "STRING"
```

### Instead, this happened

Since this is a physical name, it means that it is impossible to deploy multiple instances of the app (e.g. with `wing test`), and generally it's a hassle.

### Is there a workaround?

_No response_

### Component

SDK

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by jogold**
If a name is provided then the `cloud.Secret` references an existing secret
in the target cloud. If no name is provided then a new secret is provisioned
in the target cloud.

⚠️ It is no longer possible to provision a secret with a well-defined/well-known name.

Closes #2252 
 
*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/target-awscdk/secret.ts [hunk 1]
```diff
@@ -12,14 +15,14 @@ import { IInflightHost } from "../std";
...
 
   constructor(scope: Construct, id: string, props: cloud.SecretProps = {}) {
     super(scope, id, props);
 
-    this.secret = new CdkSecret(this, "Default", {
-      secretName: props.name,
-    });
+    this.secret = props.name
+      ? CdkSecret.fromSecretNameV2(this, "Default", props.name)
+      : new CdkSecret(this, "Default");
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-awscdk/secret.ts:25]: (not blocking) I wonder if we this is something we should mention in the docstring for `cloud.Secret`, or if this is something we could document in a dedicated README / conceptual docs page about the `cloud.Secret` resource?

**Comment by jogold** [comment]
In `sim` we could "mimic" the provisioning by creating an entry in `~/.wing/secrets.json` with `null` as the value?

File: libs/wingsdk/src/target-awscdk/secret.ts (hunk 1 above)
Comment (Chriscbr) [reply libs/wingsdk/src/target-awscdk/secret.ts]: Ah I forgot we have an issue related to this https://github.com/winglang/wing/issues/1438 - @jogold is this something you'd like to try? (Only if it interests you of course 🙂)

**Comment by Chriscbr** [comment]
> In `sim` we could "mimic" the provisioning by creating an entry in `~/.wing/secrets.json` with `null` as the value?

Sounds reasonable to me. Maybe we want to log something during preflight so the user knows that a file was added to their system where secrets should be inserted

File: libs/wingsdk/src/target-awscdk/secret.ts (hunk 1 above)
Comment (jogold) [reply libs/wingsdk/src/target-awscdk/secret.ts]: Maybe... not sure 🙂  will let know and assign myself if I do.

**Pull request merged**